libopae/src/event.c

```diff
@@ -583,6 +583,11 @@ fpga_result __FPGA_API__ fpgaCreateEventHandle(fpga_event_handle *event_handle)
 
 fpga_result __FPGA_API__ fpgaDestroyEventHandle(fpga_event_handle *event_handle)
 {
+	//sanity check
+	if (!event_handle) {
```

Comment:
This check is redundant.  It is performed by event_handle_check_and_lock().
I think the problem here is actually on line 591.  We should not be de-referencing event_handle in the assignment.  The correct assignment is:

struct _fpga_event_handle _*eh = (struct _fpga_event_handle *) event_handle;

Then, the call to event_handle_check_and_lock() on 595 will do the right thing.

@luebbers can you confirm?